src/proxy/inbound_passthrough.rs

```diff
@@ -49,7 +49,11 @@ impl InboundPassthrough {
 
         let transparent = super::maybe_set_transparent(&pi, &listener)?;
         // Override with our explicitly configured setting
-        pi.cfg.enable_original_source = Some(transparent);
+        if pi.cfg.enable_original_source.is_none() {
+            let mut cfg = (*pi.cfg).clone();
+            cfg.enable_original_source = Some(transparent);
```

Comment:
IDK why we bother to mutate the primary config with this TBH - the override is only used here. We might as well keep config R/O and set a private field, and skip cloning the whole thing.